app/main/posts/views/post-view-map.directive.js

```diff
@@ -147,54 +145,47 @@ function PostViewMap(PostEndpoint, Maps, _, PostFilters, L, $q, $rootScope, $com
         }
 
         function reloadMapPosts(query) {
-            var test = loadPosts(query);
-            test.then(addPostsToMap);
+            loadPosts(query)
         }
 
-        function loadPosts(query, offset, currentBlock) {
+        function getStats () {
             // Getting stats for filter-dropdown
             let def = PostFilters.getQueryParams(PostFilters.getDefaults());
-                PostEndpoint.geojson(def).$promise.then(res => {
-                    $scope.stats.totalItems = res.features.length;
-                    $scope.stats.unmapped = res.total - res.features.length;
+            PostEndpoint.geojson(def).$promise.then(res => {
+                $scope.stats.totalItems = res.features.length;
+                $scope.stats.unmapped = res.total - res.features.length;
             });
+        }
 
-            offset = offset || 0;
-            currentBlock = currentBlock || 1;
-
+        function loadPosts(query) {
+            let offset = 0;
+            let limit = 200;
             query = query || PostFilters.getQueryParams($scope.filters);
-
-            var conditions = _.extend(query, {
+            let conditions = _.extend(query, {
                 limit: limit,
                 offset: offset,
                 has_location: 'mapped'
             });
 
-            var request = PostEndpoint.geojson(conditions);
-            currentGeoJsonRequests.push(request);
-
-            return request.$promise.then(function (posts) {
-                // Set number of posts for filter-dropdown
+            let getFirstPostChunk = PostEndpoint.geojson(conditions);
+            currentGeoJsonRequests.push(getFirstPostChunk);
+            getFirstPostChunk.$promise.then(function (posts) {
+                // Adding the first 200 posts to map here and getting the totals
                 $scope.stats.filteredPosts = posts.total;
+                addPostsToMap(posts)
+
+                // Moving on to request rest of the posts
+                if (posts.total > limit) {
+                    for (let i = 200; i < posts.total; i = i + limit) {
```

Comment:
Instead of the literal `200`, the limit variable could be reused.

Reply:
agree. Thanks. 
@Angamanga pinging for you to follow up on this small change and then we could merge, I think?